### PR TITLE
[Feature] Admin department page styling

### LINKED ISSUE #11478: ✨ New nav - Departments
## 🛑 Blockers

```[tasklist]
### Blocked By
```

### PR DESCRIPTION
**Pull request by petertgiles**
🤖 Resolves #11478

## 👋 Introduction

Updates the styling of the admin department pages and adds a new "view department" page.

## 🕵️ Details

I've rolled a few more things into this one.  

I reduced the bottom padding on the new hero to x2.  I've also removed the built-in navbar and x3 of top padding.
image

I created a new CardSectioned component.  Visually it is based on CardBasic and it is broken up like PreviewList.  It's not very flexible right now because we didn't receive any designs for it.  Hopefully it can become a standard component that is reused widely someday.

image

## 🧪 Testing

1. Rebuild and log in as admin
2. View the department pages

- [ ] Mobile
- [ ] Desktop

## 📸 Screenshot

image

### CODE REVIEW

File: apps/web/src/pages/Departments/components/DepartmentTable.tsx [hunk 1]
```diff
@@ -95,26 +92,27 @@ export const DepartmentTable = ({
...
+          description: "Label for the keyword search input",
         }),
       }}
       add={{
         linkProps: {
           href: paths.departmentCreate(),
           label: intl.formatMessage({
-            defaultMessage: "Create Department",
-            id: "ZbpbD6",
+            defaultMessage: "Create new department",
```
Comment (mnigh) [inline apps/web/src/pages/Departments/components/DepartmentTable.tsx:104]: New is implied when creating something. Can we remove the word new?

File: apps/web/src/pages/Departments/components/DepartmentTable.tsx [hunk 2]
```diff
@@ -47,31 +46,29 @@ export const DepartmentTable = ({
...
+        isRowTitle: true,
+      },
+    }),
     columnHelper.accessor("departmentNumber", {
       id: "departmentNumber",
       filterFn: "weakEquals",
       header: intl.formatMessage({
-        defaultMessage: "Department #",
-        id: "QOvS1b",
+        defaultMessage: "Number",
```
Comment (mnigh) [inline apps/web/src/pages/Departments/components/DepartmentTable.tsx:66]: Everywhere else in the forms, it is referred to as Department number. Should it not be kept the same for consistency?

File: apps/web/src/pages/Departments/components/DepartmentTable.tsx [hunk 3]
```diff
@@ -95,26 +92,27 @@ export const DepartmentTable = ({
       search={{
         internal: true,
         label: intl.formatMessage({
-          defaultMessage: "Search departments",
-          id: "bUyxJi",
-          description: "Label for the departments table search input",
+          defaultMessage: "Search by keyword",
```
Comment (mnigh) [inline apps/web/src/pages/Departments/components/DepartmentTable.tsx:95]: What is a keyword?

File: apps/web/src/pages/Departments/components/DepartmentTable.tsx [hunk 4]
```diff
@@ -95,26 +92,27 @@ export const DepartmentTable = ({
...
           }),
           from: currentUrl,
         },
       }}
       nullMessage={{
         description: intl.formatMessage({
-          defaultMessage: 'Use the "Create Department" button to get started.',
-          id: "yat9wx",
+          defaultMessage:
+            'Use the "Create new department" button to get started.',
```
Comment (mnigh) [inline apps/web/src/pages/Departments/components/DepartmentTable.tsx:114]: Same as https://github.com/GCTC-NTGC/gc-digital-talent/pull/11853/files#diff-0fbd163c841c797ff5bf7a62a349a7d280b11674a638626e14ef6e128ecf6881R104.

**Review by mnigh — COMMENTED** [review]
Some comments related to copy.

**Review by yonikid15 — CHANGES_REQUESTED** [review]
Nice work as always :tada: 
For some reason the nav menu's z-index is overlapping the overlay only on the edit page, when the page is loading :confused: (why isn't it happening on the other pages too :laughing:). To fix this, we can create a new issue or just set them to "2" in lines [NavigationMenu-L45](https://github.com/GCTC-NTGC/gc-digital-talent/blob/11478-admin-departments-styling/apps/web/src/components/Hero/NavigationMenu.tsx#L45) and [NavigationMenu-L184](https://github.com/GCTC-NTGC/gc-digital-talent/blob/11478-admin-departments-styling/apps/web/src/components/Hero/NavigationMenu.tsx#L184)

Screenshot from 2024-10-29 14-34-27

**Comment by yonikid15** [comment]
Also the sign out and notifications buttons dark/light modes, aren't synced up with the rest of the dialog. (See far right of nav menu in image in light mode).
image

Again, this is out of scope, but seems like an easy fix. If not I can create a new issue. Otherwise everything looks good design and code wise:+1:

**Comment by petertgiles** [comment]
As discussed with Josh, I just removed the navbar that was built-in to the hero and x3 of top padding.
[remove hero nav and padding](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11853/commits/7db6149a2ba4163bf270d0abe74a9b61c0edd4de)

File: apps/web/src/pages/Departments/UpdateDepartmentPage.tsx [hunk 5]
```diff
@@ -191,12 +237,21 @@ const UpdateDepartmentPage = () => {
       return Promise.reject(new Error(result.error?.toString()));
     });
 
+  const departmentName = getLocalizedName(
+    departmentData?.department?.name,
+    intl,
+  );
+
```
[the remark this answers is not in the document]
Comment (petertgiles) [reply apps/web/src/pages/Departments/UpdateDepartmentPage.tsx:244]: This is a good catch.  I could have this be an empty string or "Loading" or something if you want.

[the remark this answers is not in the document]
Comment (petertgiles) [reply apps/web/src/pages/Departments/UpdateDepartmentPage.tsx:244]: OK, I added the loading string
[loading title for view page](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11853/commits/d881f102b23e050aa5a73566c917f2af46b80dd1)

**Pull request merged**